gpm_api/tests/test_visualization/test_plot.py

```diff
@@ -0,0 +1,334 @@
+import cartopy.crs as ccrs
+import inspect
+import os
+import pytest
+from matplotlib import image as mpl_image, pyplot as plt
+import numpy as np
+import tempfile
+import xarray as xr
+
+
+from gpm_api import _root_path
+from gpm_api.visualization import plot
+
+
+plots_dir_path = os.path.join(_root_path, "gpm_api", "tests", "data", "plots")
+image_extension = ".png"
+mse_tolerance = 5e-3
+
+
+# Utils functions ##############################################################
+
+
+def save_and_check_figure(
+    name: str,
+) -> None:
+    """Save the current figure to a temporary location and compare it to the reference figure
+
+    If the reference figure does not exist, it is created and the test is skipped.
+    """
+
+    # Save reference figure if it does not exist
+    reference_path = os.path.join(plots_dir_path, name + image_extension)
+
+    if not os.path.exists(reference_path):
+        os.makedirs(plots_dir_path, exist_ok=True)
+        plt.savefig(reference_path)
```

Comment:
I wonder if it's not better to pass the plotter object (or the figure object) returned by the plotting functions as argument to this function... and then do `plotter.figure.savefig()` or  `fig.save_fig()`. This would avoid searching in the global environment for the plot.

Example: 

```
p = plot.plot_map_mesh(grid_antimeridian_dataarray)
 save_and_check_figure(plotter=p, name=get_test_name(self))
```
or
```
p = plot.plot_map_mesh(grid_antimeridian_dataarray)
 save_and_check_figure(fig=p.figure, name=get_test_name(self))
```

Reply:
Yes that is better, less bug prone